ivy/utils/backend/sub_backend_handler.py

```diff
@@ -200,3 +209,67 @@ def find_available_sub_backends(sub_backends_loc):
             available_sub_backends.append(sub_backend)
 
     return available_sub_backends
+
+
+def _find_available_sub_backend_implementations(sub_backends):
+    result = dict()
+    for sub in sub_backends:
+        try:
+            sub_backend = ivy.utils.dynamic_import.import_module(_sub_backend_dict[sub])
+        except ModuleNotFoundError:
+            continue
+        for k, v in sub_backend.__dict__.items():
```

Comment:
I think this loop could be avoided with a single line instead, I might be missing something though

Reply:
yeah, here we search all possible sub backends for efficient implementations so we can Cache them in `_available_sub_backends_implementations_dict  `

Reply:
I think we should only search for one function at a time in the sub_backend dict particularly given that neither solution is sub-optimal in terms of time with the advantage of not having to maintain a cache dict and making debugging easier if a particular sub-backend implementation is retrieved when we try to wrap that particular function in the backend loop along with having a simpler implementation of this logic as well. What do you think?

Reply:
As I understand it: when we set the backend let's say to `torch `we have to check if there's an efficient implementation for every function and let's say that we have 10 sub-backends so if we used the cache we traverse these sub-backends once namely the first time the `_set_backend_as_ivy` tries to assign a function, for the second function we only check `cache["fn_name"] ` to give us all available sub-backends implementations for this function and so on for the rest of the functions.
but if we don't use a cache we have to loop on the 10 sub-backends for every function and I think this would become noticeable as we add more sub-backends.
But maybe there's not significant performance improvement, the only way to know for sure is to measure the two solutions, maybe I can find a way to mock the sub-backends and see the difference. What do you say ?

Reply:
Yeah I think both approaches are roughly equivalent because you'd still need to loop over all available functions when you're wrapping the first function anyway. It's just about whether we want to iterate over all functions available when we're trying to check for the first function or whether we want to check for sub-backend implementations when we iterate over those functions. Given that we'd be having a relatively small number of sub-backend functions (3 as of now) and sub-backends (2), I think both approaches would roughly be the same in terms of performance because people wouldn't really switch sub-backends that often.
I think for now we can skip the cache approach to simplify the implementation, but surely if we have a large number of sub-backends and sub-backend implementations, we could consider caching for faster retrieval 😄

Reply:
understood 😃